File: website/UserControls/SoftwareDesign_Aptify/LayoutWidgets/ApplicationWidget/Designer/ApplicationWidgetDesigner.js
Type.registerNamespace("SitefinityWebApp.UserControls.SoftwareDesign_Aptify.LayoutWidgets.ApplicationWidget.Designer");

SitefinityWebApp.UserControls.SoftwareDesign_Aptify.LayoutWidgets.ApplicationWidget.Designer.ApplicationWidgetDesigner = function (element) {
    /* Initialize fields */
    this._appOneTitle = null;
    this._appOneDate = null;
    this._appTwoTitle = null;
    this._appTwoDate = null;
    this._appThreeTitle = null;
    this._appThreeDate = null;

    /* Calls the base constructor */
    SitefinityWebApp.UserControls.SoftwareDesign_Aptify.LayoutWidgets.ApplicationWidget.Designer.ApplicationWidgetDesigner.initializeBase(this, [element]);
}

SitefinityWebApp.UserControls.SoftwareDesign_Aptify.LayoutWidgets.ApplicationWidget.Designer.ApplicationWidgetDesigner.prototype = {
    /* --------------------------------- set up and tear down --------------------------------- */
    initialize: function () {
        /* Here you can attach to events or do other initialization */
        SitefinityWebApp.UserControls.SoftwareDesign_Aptify.LayoutWidgets.ApplicationWidget.Designer.ApplicationWidgetDesigner.callBaseMethod(this, 'initialize');
    },
    dispose: function () {
        /* this is the place to unbind/dispose the event handlers created in the initialize method */
        SitefinityWebApp.UserControls.SoftwareDesign_Aptify.LayoutWidgets.ApplicationWidget.Designer.ApplicationWidgetDesigner.callBaseMethod(this, 'dispose');
    },

    /* --------------------------------- public methods ---------------------------------- */

    findElement: function (id) {
        var result = jQuery(this.get_element()).find("#" + id).get(0);
        return result;
    },

    /* Called when the designer window gets opened and here is place to "bind" your designer to the control properties */
    refreshUI: function () {
        var controlData = this._propertyEditor.get_control(); /* JavaScript clone of your control - all the control properties will be properties of the controlData too */

        /* RefreshUI  */
        jQuery(this.get_appOneTitle()).val(controlData.AppOneTitle);
        jQuery(this.get_appOneDate()).val(controlData.AppOneDate);
        jQuery(this.get_appTwoTitle()).val(controlData.AppTwoTitle);
        jQuery(this.get_appTwoDate()).val(controlData.AppTwoDate);
        jQuery(this.get_appThreeTitle()).val(controlData.AppThreeTitle);
        jQuery(this.get_appThreeDate()).val(controlData.AppThreeDate);
    },

    /* Called when the "Save" button is clicked. Here you can transfer the settings from the designer to the control */
    applyChanges: function () {
        var controlData = this._propertyEditor.get_control();

        /* ApplyChanges */
        controlData.AppOneTitle = jQuery(this.get_appOneTitle()).val();
        controlData.AppOneDate = jQuery(this.get_appOneDate()).val();
        controlData.AppTwoTitle = jQuery(this.get_appTwoTitle()).val();
        controlData.AppTwoDate = jQuery(this.get_appTwoDate()).val();
        controlData.AppThreeTitle = jQuery(this.get_appThreeTitle()).val();
        controlData.AppThreeDate = jQuery(this.get_appThreeDate()).val();
    },

    /* --------------------------------- event handlers ---------------------------------- */

    /* --------------------------------- private methods --------------------------------- */

    /* --------------------------------- properties -------------------------------------- */

    /* properties */
    get_appOneTitle: function () { return this._appOneTitle; },
    set_appOneTitle: function (value) { this._appOneTitle = value; },
    get_appOneDate: function () { return this._appOneDate; },
    set_appOneDate: function (value) { this._appOneDate = value; },
    get_appTwoTitle: function () { return this._appTwoTitle; },
    set_appTwoTitle: function (value) { this._appTwoTitle = value; },
    get_appTwoDate: function () { return this._appTwoDate; },
    set_appTwoDate: function (value) { this._appTwoDate = value; },
    get_appThreeTitle: function () { return this._appThreeTitle; },
    set_appThreeTitle: function (value) { this._appThreeTitle = value; },
    get_appThreeDate: function () { return this._appThreeDate; },
    set_appThreeDate: function (value) { this._appThreeDate = value; }
}

SitefinityWebApp.UserControls.SoftwareDesign_Aptify.LayoutWidgets.ApplicationWidget.Designer.ApplicationWidgetDesigner.registerClass('SitefinityWebApp.UserControls.SoftwareDesign_Aptify.LayoutWidgets.ApplicationWidget.Designer.ApplicationWidgetDesigner', Telerik.Sitefinity.Web.UI.ControlDesign.ControlDesignerBase);

File: website/UserControls/SoftwareDesign_Aptify/LayoutWidgets/ApplicationWidget/Designer/ApplicationWidgetDesigner.cs
using System;
using System.Linq;
using System.Web.UI;
using Telerik.Sitefinity.Web.UI;
using Telerik.Sitefinity.Web.UI.ControlDesign;
using System.Collections.Generic;

namespace SitefinityWebApp.UserControls.SoftwareDesign_Aptify.LayoutWidgets.ApplicationWidget.Designer
{
    /// <summary>
    /// Represents a designer for the <typeparamref name="SitefinityWebApp.UserControls.SoftwareDesign_Aptify.LayoutWidgets.ApplicationWidget.ApplicationWidget"/> widget
    /// </summary>
    public class ApplicationWidgetDesigner : ControlDesignerBase
    {
        #region Properties
        /// <summary>
        /// Obsolete. Use LayoutTemplatePath instead.
        /// </summary>
        protected override string LayoutTemplateName
        {
            get
            {
                return string.Empty;
            }
        }

        /// <summary>
        /// Gets the layout template's relative or virtual path.
        /// </summary>
        public override string LayoutTemplatePath
        {
            get
            {
                if (string.IsNullOrEmpty(base.LayoutTemplatePath))
                    return ApplicationWidgetDesigner.layoutTemplatePath;
                return base.LayoutTemplatePath;
            }
            set
            {
                base.LayoutTemplatePath = value;
            }
        }

        protected override HtmlTextWriterTag TagKey
        {
            get
            {
                return HtmlTextWriterTag.Div;
            }
        }
        #endregion

        #region Control references
       
        /// <summary>
        /// Gets the control that is bound to the AppOneTitle property
        /// </summary>
        protected virtual Control AppOneTitle
        {
            get
            {
                return this.Container.GetControl<Control>("AppOneTitle", true);
            }
        }
        /// <summary>
        /// Gets the control that is bound to the AppOneDate property
        /// </summary>
        protected virtual Control AppOneDate
        {
            get
            {
                return this.Container.GetControl<Control>("AppOneDate", true);
            }
        }

        protected virtual Control AppTwoTitle
        {
            get
            {
                return this.Container.GetControl<Control>("AppTwoTitle", true);
            }
        }
        /// <summary>
        /// Gets the control that is bound to the AppTwoDate property
        /// </summary>
        protected virtual Control AppTwoDate
        {
            get
            {
                return this.Container.GetControl<Control>("AppTwoDate", true);
            }
        }

        protected virtual Control AppThreeTitle
        {
            get
            {
                return this.Container.GetControl<Control>("AppThreeTitle", true);
            }
        }
        /// <summary>
        /// Gets the control that is bound to the AppThreeDate property
        /// </summary>
        protected virtual Control AppThreeDate
        {
            get
            {
                return this.Container.GetControl<Control>("AppThreeDate", true);
            }
        }

        #endregion

        #region Methods
        protected override void InitializeControls(Telerik.Sitefinity.Web.UI.GenericContainer container)
        {
            // Place your initialization logic here
            if (this.PropertyEditor != null)
            {
                var uiCulture = this.PropertyEditor.PropertyValuesCulture;
            }
        }
        #endregion

        #region IScriptControl implementation
        /// <summary>
        /// Gets a collection of script descriptors that represent ECMAScript (JavaScript) client components.
        /// </summary>
        public override IEnumerable<ScriptDescriptor> GetScriptDescriptors()
        {
            var scriptDescriptors = new List<ScriptDescriptor>(base.GetScriptDescriptors());
            var descriptor = (ScriptControlDescriptor)scriptDescriptors.Last();

            descriptor.AddElementProperty("appOneTitle", this.AppOneTitle.ClientID);
            descriptor.AddElementProperty("appOneDate", this.AppOneDate.ClientID);
            descriptor.AddElementProperty("appTwoTitle", this.AppTwoTitle.ClientID);
            descriptor.AddElementProperty("appTwoDate", this.AppTwoDate.ClientID);
            descriptor.AddElementProperty("appThreeTitle", this.AppThreeTitle.ClientID);
            descriptor.AddElementProperty("appThreeDate", this.AppThreeDate.ClientID);

            return scriptDescriptors;
        }

        /// <summary>
        /// Gets a collection of ScriptReference objects that define script resources that the control requires.
        /// </summary>
        public override System.Collections.Generic.IEnumerable<System.Web.UI.ScriptReference> GetScriptReferences()
        {
            var scripts = new List<ScriptReference>(base.GetScriptReferences());
            scripts.Add(new ScriptReference(ApplicationWidgetDesigner.scriptReference));
            return scripts;
        }
        #endregion

        #region Private members & constants
        public static readonly string layoutTemplatePath = "~/UserControls/SoftwareDesign_Aptify/LayoutWidgets/ApplicationWidget/Designer/ApplicationWidgetDesigner.ascx";
        public const string scriptReference = "~/UserControls/SoftwareDesign_Aptify/LayoutWidgets/ApplicationWidget/Designer/ApplicationWidgetDesigner.js";
        #endregion
    }
}
 
